### PR TITLE
Modify library to accept partition path

### PR DESCRIPTION
**Pull request by ryshah**
Fixing the library issue

Changes to format specific partition

Consider lvm partitions for Swift in local

Change name for OS shared partition var

Fix the syntax issues

### CODE REVIEW

File: library/swift_disk.py [hunk 1]
```diff
@@ -29,23 +38,39 @@
...
     dev = module.params.get('dev')
-    dev_path = "/dev/%s" % dev
-    part_path = "%s1" % dev_path
-
-    if not os.path.exists(dev_path):
+    part_path = module.params.get('partition_path')
+    
+    if (dev is None and part_path is None) or \
+          (dev is not None and part_path is not None) :
+        module.fail_json(msg="only one of the dev or partition_path must be set")
```
Comment (j2sol) [inline library/swift_disk.py]: There is a convenience method to express this.  See https://groups.google.com/forum/#!topic/ansible-devel/ny8YSlCWVSQ

Comment (j2sol) [inline library/swift_disk.py]: There is also a method for saying one of an option is required, `required_one_of`. See https://github.com/ansible/ansible-modules-core/blob/devel/packaging/language/pip.py#L230-L248 for example usage.

File: library/swift_disk.py [hunk 2]
```diff
@@ -4,22 +4,31 @@
...
-# backed by a file mounted over loop device
+# use the sdb disk for swift. it will be formatted to xfs
+# and mounted on /srv/node/sdb
 
 - swift_disk: dev=sdb
+
+# use the lvm disk /dev/vgpool/sftmeta. Partition will be formatted
+# and mounted on /srv/node/sftmeta
+
+- swift_disk: partition_path=/srv/node/sftmeta
```
Comment (j2sol) [inline library/swift_disk.py]: I don't understand this example. partition_path says a complete path to the partition, but the value you give here is a path to where it's to be mounted.

File: library/swift_disk.py [hunk 3]
```diff
@@ -59,21 +84,21 @@ def main():
...
     try:
         with open('/etc/fstab', 'a') as f:
-            f.write("UUID=%s /srv/node/%s1 xfs noatime,nodiratime,nobarrier,logbufs=8 0 0\n" % (fsuuid, dev))
+            f.write("UUID=%s %s xfs noatime,nodiratime,nobarrier,logbufs=8 0 0\n" % (fsuuid, mount_point))
     except Exception, e:
         module.fail_json(msg="failed to update fstab: %s" % e)
 
     # mount point
-    os.makedirs('/srv/node/%s1' % dev)
+    os.makedirs(mount_point)
```
Comment (j2sol) [inline library/swift_disk.py]: At some point we should probably remove the fstab munging bit of these libraries and use Ansible's module for this. The module here should just prep the disks to be mounted, that's easier to test for work already complete.

File: library/swift_disk.py (hunk 2 above)
Comment (ryshah) [inline library/swift_disk.py]: Sorry my bad.. example should say partition_path=/dev/vgpool/sftmeta

File: library/swift_disk.py (hunk 1 above)
Comment (ryshah) [inline library/swift_disk.py]: Thanks - this makes sense. I will change it to use required_one_of..

File: library/swift_disk.py (hunk 3 above)
Comment (ryshah) [inline library/swift_disk.py]: Agreed - you are suggesting it as future work, right?

Comment (j2sol) [inline library/swift_disk.py]: Yes, future work.

**Comment by ryshah** [comment]
@j2sol Can we get this merged?

**Pull request merged**